pyteal/ast/gitxn.py

```diff
@@ -51,6 +55,21 @@ def __init__(self, txnIndex: int, field: TxnField, index: Union[int, Expr]) -> N
         super().__init__(Op.gitxna, Op.gitxnas, "Gitxna", field, index)
         self.txnIndex = txnIndex
 
+        def validate_types_or_throw():
+            is_index_expr = isinstance(self.index, Expr)
```

Comment:
I think the `TxnaExpr` constructor should check `index`, so this `GitxnaExpr ` should only have to worry about checking `txnIndex`. The same thing applies to `GtxnaExpr`

Reply:
@jasonpaulos Via https://github.com/michaeldiamant/pyteal/pull/1/commits/4409cd13a281040108718a66a0f6cf71b26a38e9, I attempted to move `index` validation.

The changes sacrifices visibility of all parameters (e.g. `txnIndex`) in error messages.  Since other validation happens in the parent class, I assume the tradeoff is considered _reasonable_.